File: first/HelloWorld.java
import java.util.Arrays;

public class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello, World!");

        int[] array = {5, 2, 9, 1, 5, 6};
        System.out.println("Original Array: " + Arrays.toString(array));

        bubbleSort(array);
        System.out.println("Bubble Sorted Array: " + Arrays.toString(array));

        int[] array2 = {3, 7, 4, 9, 5, 2};
        selectionSort(array2);
        System.out.println("Selection Sorted Array: " + Arrays.toString(array2));

        int[] array3 = {8, 3, 1, 7, 0, 10, 2};
        insertionSort(array3);
        System.out.println("Insertion Sorted Array: " + Arrays.toString(array3));
    }

    public static void bubbleSort(int[] array) {
        int n = array.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
    }

    public static void selectionSort(int[] array) {
        int n = array.length;
        for (int i = 0; i < n - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < n; j++) {
                if (array[j] < array[minIdx]) {
                    minIdx = j;
                }
            }
            int temp = array[minIdx];
            array[minIdx] = array[i];
            array[i] = temp;
        }
    }

    public static void insertionSort(int[] array) {
        int n = array.length;
        for (int i = 1; i < n; ++i) {
            int key = array[i];
            int j = i - 1;

            while (j >= 0 && array[j] > key) {
                array[j + 1] = array[j];
                j = j - 1;
            }
            array[j + 1] = key;
        }
    }
}

File: first/helloWorld.c
#include <stdio.h>

// Function prototypes for sorting algorithms
void bubbleSort(int arr[], int n);
void selectionSort(int arr[], int n);
void insertionSort(int arr[], int n);
void printArray(int arr[], int size);

int main() {
    printf("Hello, World!\n");

    int arr[] = {64, 25, 12, 22, 11};
    int n = sizeof(arr)/sizeof(arr[0]);

    printf("Original array: \n");
    printArray(arr, n);

    bubbleSort(arr, n);
    printf("Sorted array with Bubble Sort: \n");
    printArray(arr, n);

    // Reset array to original unsorted state
    int arr2[] = {64, 25, 12, 22, 11};
    selectionSort(arr2, n);
    printf("Sorted array with Selection Sort: \n");
    printArray(arr2, n);

    // Reset array to original unsorted state
    int arr3[] = {64, 25, 12, 22, 11};
    insertionSort(arr3, n);
    printf("Sorted array with Insertion Sort: \n");
    printArray(arr3, n);

    return 0;
}

void bubbleSort(int arr[], int n) {
    int i, j;
    for (i = 0; i < n-1; i++) {
        for (j = 0; j < n-i-1; j++) {
            if (arr[j] > arr[j+1]) {
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}

void selectionSort(int arr[], int n) {
    int i, j, min_idx;
    for (i = 0; i < n-1; i++) {
        min_idx = i;
        for (j = i+1; j < n; j++) {
            if (arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        int temp = arr[min_idx];
        arr[min_idx] = arr[i];
        arr[i] = temp;
    }
}

void insertionSort(int arr[], int n) {
    int i, key, j;
    for (i = 1; i < n; i++) {
        key = arr[i];
        j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = key;
    }
}

void printArray(int arr[], int size) {
    int i;
    for (i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}
